### PR TITLE
perf(collaboration): add telemetry, log, update warning message

### PR DESCRIPTION
**Pull request by SLdragon**

### CODE REVIEW

File: packages/vscode-extension/src/handlers.ts [hunk 1]
```diff
@@ -847,18 +847,33 @@ export async function listCollaborator(env: string): Promise<TreeItem[]> {
...
           commandId: `fx-extension.listcollaborator.${env}`,
-          label: "No permission to list all collaborators.",
+          label: "No permission to list collaborators",
           icon: "warning",
           isCustom: true,
           parent: "fx-extension.environment." + env,
         },
       ];
     }
+    ExtTelemetry.sendTelemetryEvent(Stage.listCollaborator, {
```
Comment (KennethBWSong) [inline packages/vscode-extension/src/handlers.ts:857]: Also send start telemetry at the beginning？

Comment (SLdragon) [reply packages/vscode-extension/src/handlers.ts:857]: OK, thanks for the tip

File: packages/vscode-extension/src/handlers.ts [hunk 2]
```diff
@@ -892,7 +907,11 @@ export async function checkPermission(env: string): Promise<boolean> {
     result =
       (teamsAppPermission.roles?.includes("Administrator") ?? false) &&
       (aadPermission.roles?.includes("Owner") ?? false);
+    ExtTelemetry.sendTelemetryEvent(Stage.checkPermission, {
```
Comment (KennethBWSong) [inline packages/vscode-extension/src/handlers.ts:910]: same as above. Also another question, why telemetry is not sent in grant permission?

Comment (KennethBWSong) [reply packages/vscode-extension/src/handlers.ts:910]: By the way, seems log is not added?

Comment (SLdragon) [reply packages/vscode-extension/src/handlers.ts:910]: processResult function at end will send the telemetry and show error dialog, so not send here

File: packages/vscode-extension/src/handlers.ts (hunk 1 above)
Comment (SLdragon) [reply packages/vscode-extension/src/handlers.ts]: Updated

**Review by KennethBWSong — APPROVED** [review]

**Pull request merged**